recipes/_linux.rb

```diff
@@ -89,6 +89,33 @@
     flush_cache node['sensu']['yum_flush_cache'] unless node['sensu']['yum_flush_cache'].nil?
     notifies :create, "ruby_block[sensu_service_trigger]"
   end
+when "amazon"
+  repo = yum_repository "sensu" do
+    description "sensu monitoring"
+    repo = node["sensu"]["use_unstable_repo"] ? "yum-unstable" : "yum"
+    # Use the platform version numbering scheme to determine the release version
+    # 20XX.XX vs 2.X
+    releasever_string = node["sensu"]["yum_repo_releasever"] || /^20\d\d/.match(node["platform_version"]) ? "6" : "7"
```

Comment:
A couple of thoughts:
- I will check whats the minimum version of ruby we support (based on the version of `chef-client`, if we have a new enough we can use `.match?(//)` which is more performant because it discards the actual match data and just stores a boolean value.
- I think it would be safer to use `/201\d/` as a regex as it limits the scope to known releases and should amazon decide in a couple years to go back to this style of versioning then it would impact us
- I am thinking this really should be a case statement and in the event that it does not match `2` or the yearly regex we should bail